Remove Disroot email recommendation

jonaharagon opened the pull request:
Unfortunately, Disroot does not meet our minimum criteria for recommendation:

- Protection of webmail with 2FA, such as TOTP. (#1075)

I considered removing our 2FA requirement instead of this PR, as it was pointed out that webmail 2FA provides little additional security when IMAP/SMTP does not allow for two-factor authentication anyways. However, Disroot also does not meet a number of our best practices for recommendation, including:

- Encrypts email account data at rest.
- Support for hardware authentication, ie U2F and [WebAuthn](https://en.wikipedia.org/wiki/WebAuthn).
- Offering email aliases. (#1109)
- Accepting Monero for purchases.

Additionally, removing Disroot allows us to increase our minimum security requirement for email storage from "encrypted" to "encrypted with zero-access encryption", a change which has also been made in this PR. Every other recommended provider utilizes (or in the case of mailbox.org, *can optionally utilize*) zero-knowledge encryption for email storage.

In light of all of this, I think the best course of action is to not recommend their email services. Ultimately I don't think Disroot provides a _bad_ service, I just believe our four other recommendations are all objectively _better_ from a privacy and security standpoint.

Closes: #1109 Closes: #1075

freddy-m reviewed (approved):
LGTM but I'll let others check over this before merging

dngray reviewed (dismissed):
> - Encrypts account data at rest.

This was always filesystem/full disk encryption, and not specifically on a per email/inbox basis. This was the reason why Mailfence was removed: https://github.com/privacytools/privacytools.io/issues/1908 I emailed them on the 9 Mar 2022 for an update and they said it still wasn't ready yet.

> - Support for hardware authentication, ie U2F and [WebAuthn](https://en.wikipedia.org/wiki/WebAuthn).

@muppeth [mentioned](https://github.com/privacyguides/privacyguides.org/issues/1075#issuecomment-1106814174) they switched to SnappyMail rather quickly because of a security issue with Rainloop. There is a plugin for 2FA with SnappyMail (apparently according to this page https://snappymail.eu/comparison.html), so I wanted to know if there was any plan to use that.

We do allow providers who allow you to disable IMAP/SMTP and use webmail only as with the situation of Mailbox.org mentioned here: https://github.com/privacyguides/privacyguides.org/issues/1075#issuecomment-1106835446 StartMail uses a similar approach, to Mailbox.org with a "[trusted IMAP password](https://support.startmail.com/hc/en-us/articles/360006536217-Configure-a-trusted-device-IMAP-password)".

I'm overly filled confidence from the author of SnappyMail describing 2FA+IMAP as "fake security", https://github.com/the-djmaze/snappymail/issues/84#issuecomment-817852617. 

There are ways to provide 2FA and also IMAP support, SASL XOAUTH2 etc. It would be nice to see more providers support this method as many clients have implemented it, because the large providers, (Google Outlook, Yahoo), already do use it.

One point that I have considered is perhaps we should adjust the 2FA requirement to be not necessary. The reason is currently all providers that provide 2FA either require you to use their app, (tutanota) bridge, (protonmail, and you must give them an encrypted copy of your PGP key) or webmail (mailbox.org).

We do know these approaches do come at the cost of *some security* as discussed in https://github.com/privacyguides/privacyguides.org/pull/1052, so by requiring MFA we may actually be recommending less secure approaches.

In other words you're trading authentication security for private key security.

> - Offering email aliases. (#1109)

This was always a "best case" scenario https://www.privacyguides.org/email/#technology and never something we required. It depends on a person's threat model, because you can combine your service with a [cloaking service](https://www.privacyguides.org/email/#email-cloaking-services) to achieve privacy.

> Accepting Monero for purchases.

They do accept Monero https://disroot.org/en/cryptocurrency.

jonaharagon commented:
I think I addressed all the points you mentioned in the original post. As far as Monero:

<img width="501" alt="CleanShot 2022-04-28 at 08 03 49@2x" src="https://user-images.githubusercontent.com/3637842/165758364-620c6986-513b-4132-99b6-b9d511ab72c3.png">

https://disroot.org/en/forms/domain-linking-form

> [...] removing Disroot allows us to increase our minimum security requirement for email storage from "encrypted" to "encrypted with zero-access encryption", a change which has also been made in this PR.

I think this is the most important change here, this should probably be a mandatory feature given the number of providers that support this functionality.

dngray commented:
> > [...] removing Disroot allows us to increase our minimum security requirement for email storage from "encrypted" to "encrypted with zero-access encryption", a change which has also been made in this PR.
> 
> I think this is the most important change here, this should probably be a mandatory feature given the number of providers that support this functionality.

That I am inclined to agree with. I did check with them and apparently they are looking at site-wide MFA of some kind, so that would be cool. Obviously they said that will take some time.

I think for now we might remove it, and then re-add it when they've done that.

dngray commented on docs/email.en.md:
```diff
@@ -351,9 +308,8 @@ Email servers deal with a lot of very sensitive data. We expect that providers w
-
-- [HTTP Strict Transport Security](https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security)
-- [Subresource Integrity](https://en.wikipedia.org/wiki/Subresource_Integrity) if loading things from external domains.
+    - [HTTP Strict Transport Security](https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security)
+    - [Subresource Integrity](https://en.wikipedia.org/wiki/Subresource_Integrity) if loading things from external domains.
```
I did spot these linter errors:

docs/email.en.md:311 [markdownlint]  Warning MD007/ul-indent: Unordered list indentation [Expected: 2; Actual: 4]
docs/email.en.md:312 [markdownlint]  Warning MD007/ul-indent: Unordered list indentation [Expected: 2; Actual: 4]

dngray commented on docs/email.en.md:
```diff
@@ -363,9 +319,8 @@ Email servers deal with a lot of very sensitive data. We expect that providers w
-
-- [Content Security Policy (CSP)](https://en.wikipedia.org/wiki/Content_Security_Policy)
-- [Expect-CT](https://datatracker.ietf.org/doc/draft-ietf-httpbis-expect-ct)
+    - [Content Security Policy (CSP)](https://en.wikipedia.org/wiki/Content_Security_Policy)
+    - [Expect-CT](https://datatracker.ietf.org/doc/draft-ietf-httpbis-expect-ct)
```
I spotted a couple more:

docs/email.en.md:322 [markdownlint]  Warning MD007/ul-indent: Unordered list indentation [Expected: 2; Actual: 4]
docs/email.en.md:323 [markdownlint]  Warning MD007/ul-indent: Unordered list indentation [Expected: 2; Actual: 4]

Weirdly doesn't seem to work if you have any less spaces.

jonaharagon replied:
> Weirdly doesn't seem to work if you have any less spaces.

Oh yeah I noticed this as well, I'm not sure why this is the case, I need to look into this.
